### PR TITLE
Modal component

### PR DESCRIPTION
**Pull request by habin-isa**
## Modal component added
- Takes title, children, handleClick (for toggle) and showModal
- Storybook example contains a button to show usage
- ⭐️ Adds drawer prop for drawer state on mobile view (Jason requested for this to be an optional prop)

image
image
image

### CODE REVIEW

File: src/Modal/Modal.stories.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    </Box>
+  )
+}
+
+export const Default = Template.bind({})
+
+Default.args = {
+  title: "Hello world i'm a beautiful modal",
+  showModal: false,
+}
```
Comment (alexwllms) [inline src/Modal/Modal.stories.js:34]: Still not adding a newline on save? VS code settings? Prettier settings?

File: src/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+export const Modal: FC<ModalProps> = ({
+  title = '',
+  body = '',
+  children = '',
+  showModal = false,
+  handleClick,
+}) => {
+
+  return (
```
Comment (alexwllms) [inline src/Modal/Modal.tsx:30]: If there's no body to the modal function can we remove the empty line and the return keyword and use the shorthand `}) => ...`

File: src/Modal/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    <Wrapper showModal={showModal}>
+      <Overlay />
+      <Container>
+        <Box flex alignItems="center" justifyContent="space-between">
+          <Text tag="h2" typo="header-small" align="left">{title}</Text>
+          <IconContainer onClick={handleClick}>
```
Comment (alexwllms) [inline src/Modal/Modal.tsx:36]: Does Jason want us to use the new icon button with this modal (seen here https://www.figma.com/file/FqSaizGOyOzXFZNmZ4X0LGMn/%F0%9F%8D%AC-S-mores?node-id=2790%3A0)?

File: src/Modal/__tests__/Modal.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import {render} from '@testing-library/react';
```
Comment (alexwllms) [inline src/Modal/__tests__/Modal.js:2]: Prettier doesn't seem to be adding spaces around the imports inside the `{ render }`

File: src/Modal/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  body?: string,
+  children?: ReactNode,
+  showModal?: boolean,
+  handleClick: () => void
+}
+
+export const Modal: FC<ModalProps> = ({
+  title = '',
+  body = '',
+  children = '',
```
Comment (alexwllms) [inline src/Modal/Modal.tsx:25]: Should children be optional? I can see the modal having an optional title but I imagine it'll always required some child elements?

File: src/Modal/Modal.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import { Text } from '../Text'
+
+interface IModal {
+  // showModal state
+  showModal: boolean
+}
+
+type ModalProps = {
+  title?: string,
+  body?: string,
```
Comment (alexwllms) [inline src/Modal/Modal.tsx:16]: Where is the body used in the component? Am I being dumb 😆 I can't see it.

**Review by alexwllms — CHANGES_REQUESTED** [review]

File: src/Modal/Modal.stories.js (hunk 1 above)
Comment (habin-isa) [reply src/Modal/Modal.stories.js:34]: Will have to check this!

File: src/Modal/Modal.tsx (hunk 3 above)
Comment (habin-isa) [reply src/Modal/Modal.tsx:36]: Discussing now 😄

File: src/Modal/Modal.tsx (hunk 6 above)
Comment (habin-isa) [reply src/Modal/Modal.tsx:16]: Oh no! Forgot to remove, me being dumb!

File: src/Modal/Modal.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  showModal: boolean
+}
+interface IModalContainer {
+  // drawer state
+  drawer: boolean
+}
+
+type ModalProps = {
+  title?: string,
+  children?: ReactNode,
```
Comment (alexwllms) [inline src/Modal/Modal.tsx:20]: Should children be optional? It's optional here but has no default state when destructured in the component?

**Review by alexwllms — APPROVED** [review]

**Comment by alexwllms** [comment]
@habin-isa one comment on the PR worth considering but then merge when you're ready!

**Pull request merged**